### PR TITLE
[TK-06619] Remote signal

### PR DESCRIPTION
**Pull request by freesig**
# Remote Signal
Adds a `remote_signal` function that remote calls a `recv_remote_signal` callback on the same zome.

## Todo:
- [x] docs
- [x] merge [wasmer](https://github.com/holochain/holochain-wasmer/pull/45) 
- [x] remove wasmer patch
- [x] Add to changelog
- [x] Remove safe block on as it's spawned anyway.

## Out of scope future items:
-  We can use capabilities to allow the signal to be emitted without needing the callback.
- Allow calling other zomes.
- Make a more general send and forget remote call host function.

## Docs copied from the hdk function:

Send a signal to a list of other agents.
This will send the data as an [AppSignal] to
this zome for all the agents supplied.

### Non-blocking
This is a non-blocking call and will not return an
error if the calls fail. This is designed to be used
as a send and forget operation.
A log will be produced at `[remote_signal]=info` if the calls
fail though (this may be removed in the future).

### Usage
Currently this requires the function `recv_remote_signal` be
exposed by this zome with a signature like:
```rust
#[hdk_extern]
fn recv_remote_signal(signal: AppSignal) -> ExternResult<()> {
    emit_signal(&signal)?;
    Ok(())
}
```
This function will also need to be added to your init as a
unrestricted cap grant so it can be called remotely.

This requirements will likely be removed in the future as
we design a better way to grant the capability to remote signal.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [workspace]
 members = [
-  # "crates/admin",
+  "crates/admin",
```
Comment (zo-el) [inline Cargo.toml:3]: Seems like there is no admin crate.

Comment (freesig) [reply Cargo.toml:3]: oh my bad

File: crates/holochain/src/test_utils/test_conductor/test_handle.rs [hunk 2]
```diff
@@ -64,6 +70,7 @@ impl TestConductorHandle {
                 })
                 .collect();
             self.0
+                 .0
```
Comment (neonphog) [inline crates/holochain/src/test_utils/test_conductor/test_handle.rs:73]: haha, good ole' rust... i wonder if tuple structs were a language design mistake

**Review by neonphog — APPROVED** [review]
Yeah, I think we'll want to iterate on this api a bit, but as an interim step, lgtm

**Comment by pjkundert** [comment]
Hey, @freesig -- this pull request results in both holochain-serialization 0.0.45 *and* 0.0.45 being included in Cargo.lock, because freesig/observability 0.1.0 has a fixed dependency in its Cargo.lock to =0.0.45.

Perhaps merge the https://github.com/pjkundert/observability `update-syn` branch in Pull Request https://github.com/freesig/observability/pull/1, and then update holochain/holochain to use the new observability?  This will prevent holochain from pulling in both versions of holochain-serialization...

**Pull request merged**